adapters/epom/epom.go

```diff
@@ -0,0 +1,172 @@
+package epom
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type EpomAdapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Epom adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &EpomAdapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *EpomAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) (requests []*adapters.RequestData, errors []error) {
+	rq, errs := a.makeRequest(request)
+
+	if len(errs) > 0 {
+		errors = append(errors, errs...)
+		return nil, errors
+	}
+
+	if rq != nil {
+		requests = append(requests, rq)
+	}
+
+	return requests, errors
+}
+
+func (a *EpomAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb.Imp
+
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impressions in request",
+		}}
+	}
+
+	if request.Device == nil || request.Device.IP == "" {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No ip address in request",
+		}}
+	}
+
+	if request.Site == nil && request.App == nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Site or app field shouldn't be empty",
+		}}
+	}
+
+	if request.Site != nil && request.App != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Site and app can't be filled simultaneously",
+		}}
+	}
```

Comment:
The site / app check is performed by PBS-Core. Both `request.Site == nil && request.App == nil` and `request.Site != nil && request.App != nil` do not need to be repeated here.

Reply:
Removed